File: backend/app/schemas/availability.py
"""稼働可否スケジュールスキーマ"""
from pydantic import BaseModel
from app.schemas.employee import EmployeeListItem


class AvailabilityMonth(BaseModel):
    month: str       # "2026-03"
    status: str      # FREE | PARTIAL | BUSY
    allocation: int
    project_name: str | None = None


class EmployeeAvailability(BaseModel):
    employee: EmployeeListItem
    months: list[AvailabilityMonth]


class AvailabilityResponse(BaseModel):
    months_header: list[str]
    items: list[EmployeeAvailability]

File: frontend/src/api/availability.ts
/**
 * 空きカレンダー API クライアント
 *
 * エンジニアの月別稼働状況を取得する。
 */
import apiClient from './client'
import type { EmployeeListItem } from './employees'

export interface AvailabilityMonth {
  month: string
  status: 'FREE' | 'PARTIAL' | 'BUSY'
  allocation: number
  project_name: string | null
}

export interface EmployeeAvailability {
  employee: EmployeeListItem
  months: AvailabilityMonth[]
}

export interface AvailabilityResponse {
  months_header: string[]
  items: EmployeeAvailability[]
}

export interface AvailabilityParams {
  months?: number
  offset_months?: number
  location?: string
  search?: string
  status?: string
}

export const availabilityApi = {
  get: (params?: AvailabilityParams) =>
    apiClient.get<AvailabilityResponse>('/availability', { params }).then(r => r.data),
}
